File: BarberMe/BarberMe.Database/Models/Recommendation.cs
namespace BarberMe.Database.Models
{
    public class Recommendation
    {
        public int RecommendationId { get; set; }

        public int UserId { get; set; }
        public User User { get; set; } = null!;

        public int ServiceId { get; set; }
        public Service Service { get; set; } = null!;

        public decimal Score { get; set; }
        public string Explanation { get; set; } = string.Empty;

        public bool? WasAccepted { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<RecommendationFeedback> Feedbacks { get; set; } = new List<RecommendationFeedback>();
    }
}

File: barberme_mobile/lib/models/recommendation.dart
class Recommendation {
  final int recommendationId;
  final int barberServiceId;
  final int serviceId;
  final String serviceName;
  final int barberId;
  final String barberName;
  final double price;
  final int durationMinutes;
  final double score;
  final String explanation;
  final bool? wasAccepted;
  final DateTime createdAt;

  Recommendation({
    required this.recommendationId,
    required this.barberServiceId,
    required this.serviceId,
    required this.serviceName,
    required this.barberId,
    required this.barberName,
    required this.price,
    required this.durationMinutes,
    required this.score,
    required this.explanation,
    required this.wasAccepted,
    required this.createdAt,
  });

  factory Recommendation.fromJson(
    Map<String, dynamic> json,
  ) {
    return Recommendation(
      recommendationId:
          json['recommendationId'] as int,
      barberServiceId:
          json['barberServiceId'] as int,
      serviceId: json['serviceId'] as int,
      serviceName:
          json['serviceName']?.toString() ?? '',
      barberId: json['barberId'] as int,
      barberName:
          json['barberName']?.toString() ?? '',
      price:
          (json['price'] as num?)?.toDouble() ?? 0,
      durationMinutes:
          json['durationMinutes'] as int? ?? 0,
      score:
          (json['score'] as num?)?.toDouble() ?? 0,
      explanation:
          json['explanation']?.toString() ?? '',
      wasAccepted:
          json['wasAccepted'] as bool?,
      createdAt: DateTime.parse(
        json['createdAt'].toString(),
      ),
    );
  }
}